src/server.ts

```diff
@@ -0,0 +1,127 @@
+    } else if (method === "GET" && url?.startsWith("/api/users/")) {
+      const userId = url.split("/").pop();
+      if (userId && isValidUUID(userId)) {
+        const user = getUserById(userId);
+        if (user) {
+          res.statusCode = 200;
+          res.end(JSON.stringify(user));
+        } else {
+          res.statusCode = 404;
+          res.end(JSON.stringify({ error: "Record not found" }));
+        }
+      } else {
+        res.statusCode = 400;
+        res.end(JSON.stringify({ error: "Invalid userId" }));
+      }
+    } else if (method === "POST" && url === "/api/users") {
+      if (body && body.username && body.age) {
+        const user = createUser(body.username, body.age, body.hobbies);
+        res.statusCode = 201;
+        res.end(JSON.stringify(user));
+      } else {
+        res.statusCode = 400;
+        res.end(JSON.stringify({ error: "Missing required fields" }));
+      }
+    } else if (method === "PUT" && url?.startsWith("/api/users/")) {
+      const userId = url.split("/").pop();
+      if (userId && isValidUUID(userId)) {
+        if (body && (body.username || body.age || body.hobbies)) {
+          const updatedUser = updateUser(userId, body);
+          if (updatedUser) {
+            res.statusCode = 200;
+            res.end(JSON.stringify(updatedUser));
+          } else {
+            res.statusCode = 404;
+            res.end(JSON.stringify({ error: "Record not found" }));
+          }
+        } else {
+          res.statusCode = 400;
+          res.end(JSON.stringify({ error: "Missing fields to update" }));
+        }
+      } else {
+        res.statusCode = 400;
+        res.end(JSON.stringify({ error: "Invalid userId" }));
+      }
+    } else if (method === "DELETE" && url?.startsWith("/api/users/")) {
+      const userId = url.split("/").pop();
+      if (userId && isValidUUID(userId)) {
+        const deleted = deleteUser(userId);
+        if (deleted) {
+          res.statusCode = 204;
+          res.end();
+        } else {
+          res.statusCode = 404;
+          res.end(JSON.stringify({ error: "Record not found" }));
+        }
+      } else {
+        res.statusCode = 400;
+        res.end(JSON.stringify({ error: "Invalid userId" }));
+      }
+    } else {
+      res.statusCode = 404;
+      res.end(JSON.stringify({ error: "Not found" }));
+    }
+  } catch (error) {
+    res.statusCode = 500;
+    res.end(JSON.stringify({ error: "Internal server error" }));
+    errorHandler(error);
+  }
+};
+
+if (!process.env.USE_CLUSTER) {
+  const server = createServer(handleRequest);
+
+  const port: number = parseInt(process.env.PORT as string, 10) || 4000;
+
+  server.listen(port, () => {
+    console.log(`Server ${process.pid} is running on port ${port}`);
+  });
+} else {
+  const numWorkers = os.cpus().length - 1;
+  const basePort: number = parseInt(process.env.PORT as string, 10) || 4000;
+
+  if (cluster.isPrimary) {
+    for (let i = 0; i < numWorkers; i++) {
```

Comment:
Unfortunately, your load balancer is not implemented. You have fork processes running starting at port 4000+. And in fact, one single fork at 4000 fulfills all requests. It was necessary to establish the server on port 4000 in the `cluster.isPrimary` block and redirect requests from it to 4001-4007 (or whatever forks we have, and forward their responses to the main port 4000.

Reply:
And here another problem begins - how to store data between processes. Two ways:
1) Send messages between processes and store data only in `cluster.isPrimary`.
2) Send http requests to the main port from the child ones, using it as a database.